tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs

```diff
@@ -88,31 +89,21 @@ internal void SetupMocks(MockHttpManager httpManager)
         [Description("Test for client assertion with X509 public certificate using sendCertificate")]
         public async Task JsonWebTokenWithX509PublicCertSendCertificateTestAsync()
         {
-            using (var httpManager = new MockHttpManager())
+            using (var harness = new MockHttpAndServiceBundle())
             {
-                var serviceBundle = ServiceBundle.CreateWithCustomHttpManager(httpManager);
-
-                SetupMocks(httpManager);
+                SetupMocks(harness.HttpManager);
                 var certificate = new X509Certificate2(
                     ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
                     MsalTestConstants.DefaultPassword);
 
-                var clientAssertion = new ClientAssertionCertificate(certificate);
-                var clientCredential = new ClientCredential(clientAssertion);
-                var app = new ConfidentialClientApplication(
-                    serviceBundle,
-                    MsalTestConstants.ClientId,
-                    ClientApplicationBase.DefaultAuthority,
-                    MsalTestConstants.RedirectUri,
-                    clientCredential,
-                    _cache,
-                    _cache)
-                {
-                    ValidateAuthority = false
-                };
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .AddKnownAuthority(
+                                                                  new System.Uri(ClientApplicationBase.DefaultAuthority),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithX509Certificate2(certificate).BuildConcrete();
```

Comment:
>WithX509Certificate2 [](start = 63, length = 20)

this looks ugly to me. can we change the name to something more generic?

Reply:
I'll talk with jm on this with the public api surface review.  i've added this to the discussion list.

---
In reply to: [247743987](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/786#discussion_r247743987) [](ancestors = 247743987)